src/Compilers/CSharp/Test/Symbol/Symbols/PartialPropertiesTests.cs

```diff
@@ -3017,11 +3017,1028 @@ partial class C
+                class A(int i) : Attribute { }
+                [AttributeUsage(AttributeTargets.All, AllowMultiple = true)]
+                class B(int i) : Attribute { }
+
+                partial class C
+                {
+                    [A(2), B(2)]
+                    public partial int this[int i] { get => 1; set { } }
+
+                    [A(1), B(1)]
+                    public partial int this[int i] { get; set; }
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics();
+
+            var indexer = (SourcePropertySymbol)comp.GetMember<NamedTypeSymbol>("C").Indexers.Single();
+            AssertEx.Equal(["A(1)", "B(1)", "A(2)", "B(2)"], indexer.GetAttributes().ToStrings());
+            AssertEx.Equal(["A(1)", "B(1)", "A(2)", "B(2)"], indexer.PartialImplementationPart!.GetAttributes().ToStrings());
+        }
+
+        [Fact]
+        public void Attributes_IndexerParameter()
+        {
+            // Unlike other symbol kinds, for parameters, the implementation part attributes are emitted first, then the definition part attributes.
+            // This is consistent with partial methods.
+            var source = """
+                #pragma warning disable CS9113 // Primary constructor parameter is unread
+
+                using System;
+                
+                [AttributeUsage(AttributeTargets.All, AllowMultiple = true)]
+                class A(int i) : Attribute { }
+                [AttributeUsage(AttributeTargets.All, AllowMultiple = true)]
+                class B(int i) : Attribute { }
+
+                partial class C
+                {
+                    public partial int this[[A(2), B(2)] int i] { get => 1; set { } }
+
+                    public partial int this[[A(1), B(1)] int i] { get; set; }
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics();
+
+            var indexer = (SourcePropertySymbol)comp.GetMember<NamedTypeSymbol>("C").Indexers.Single();
+
+            verify(indexer.Parameters.Single());
+            verify(indexer.GetMethod!.Parameters.Single());
+            verify(indexer.SetMethod!.Parameters[0]);
+
+            verify(indexer.PartialImplementationPart!.Parameters.Single());
+            verify(indexer.PartialImplementationPart!.GetMethod!.Parameters.Single());
+            verify(indexer.PartialImplementationPart!.SetMethod!.Parameters[0]);
+
+            void verify(ParameterSymbol param)
+            {
+                AssertEx.Equal(["A(2)", "B(2)", "A(1)", "B(1)"], param.GetAttributes().ToStrings());
+            }
+        }
+
+        [Fact]
+        public void Attributes_Property_DisallowedDuplicates()
+        {
+            var source = """
+                using System;
+
+                class Attr : Attribute { }
+
+                partial class C
```

Comment:
To make the multi-errors on a line more readable, consider putting the locations where there are errors on their own lines.